src/js_realm.hpp

```diff
@@ -1382,41 +1382,151 @@ void RealmClass<T>::compact(ContextType ctx, ObjectType this_object, Arguments&
+            }
 
-        return;
-    }
+            output_config = Value::validated_to_object(ctx, firstParamValue);
 
-    // enryption key is specified
-    ValueType encryption_key_arg = args[1];
+            // make sure that `path` property exists and that it is a string
+            ValueType pathValue = Object::get_property(ctx, output_config, "path");
+            if (Value::is_string(ctx, pathValue)) {
+                output_path = Value::to_string(ctx, pathValue);
+            }
+            else if (!Value::is_valid(pathValue)) {
+                throw std::runtime_error("`path` property must exist in output configuration");
+            }
+            else {
+                throw std::runtime_error("`path` property must be a string");
+            }
+
+            // check whether encryption key property exists, and whether it's a string
+            ValueType encKeyValue = Object::get_property(ctx, output_config, "encryptionKey");
+            // `encryptionKey` is optional..
+            if (Value::is_valid(encKeyValue)) {
+                if (Value::is_string(ctx, encKeyValue)) {
+                    copy_mode |= COPYMODE_ENCRYPTED;
+                    encryption_key = Value::to_binary(ctx, encKeyValue);
+                }
+            }
+            else {
+                throw std::runtime_error("encryptionKey property must be a Binary");
+            }
 
-    if (!Value::is_binary(ctx, encryption_key_arg)) {
-        throw std::runtime_error("Encryption key for 'writeCopyTo' must be a Binary.");
+            // check whether a sync config exists -- it is optional
+            ValueType syncConfigValue = Object::get_property(ctx, output_config, "sync");
+            if (Value::is_valid(syncConfigValue) && Value::is_object(ctx, syncConfigValue)) {
+                copy_mode |= COPYMODE_TOSYNC;
+            }
+            else {
+                copy_mode |= COPYMODE_TOLOCAL;
+            }
+        } break;
+
+        case COPYMODE_WITHPATH: {
+            if (args.count > 2) {
+                throw std::runtime_error(
+                    "`writeCopyTo(<path>, [encryption key])` accepts no more than two parameters");
+            }
+            copy_mode |= COPYMODE_TOLOCAL;
+
+            // make sure that `path` parameter exists and that it is a string
+            ValueType pathValue = args[0];
+            if (Value::is_string(ctx, pathValue)) {
+                output_path = Value::to_string(ctx, pathValue);
+            }
+            else if (!Value::is_valid(pathValue)) {
+                throw std::runtime_error(
+                    "`path` parameter must be first parameter to `writeCopyTo(<path>, [encryption key])`");
+            }
+            else {
+                throw std::runtime_error("`path` parameter must be a string");
+            }
+
+            if (args.count == 2) {
+                // a second parameter is given -- it must be an encryption key for the destination Realm
```

Comment:
Why does this act differently to the configuration copy mode?  Couldn't this also transfer a string to binary?

Reply:
I'm not sure what you mean.  The configuration copy mode only takes a single parameter to `writeCopyTo`

Reply:
The configuration mode takes the `encryptedKey` value from the configuration, tests if it is a string and then transforms it into a binary. (lines 1460 to 1463).
If the second argument was a string instead of a ArrayBuffer, then couldn't we also convert the argument into a binary instead of throwing an exception?